contents/handbook/engineering/development-process.md

```diff
@@ -17,7 +17,37 @@ However, if we have no checks in place, we will release quickly but everything w
 
 Deciding what we work on happens in our sprint planning session, where the high level themes for the next sprint get decided. Each team then takes those high level priorities and breaks them down, plus adds their own smaller items like bugfixes, smaller features or other improvements.
 
-As one of our values is [stepping on toes](/handbook/company/values#step-on-toes), during the sprint you might come across something that should be much higher priority than what was already planned. It's up to you to then decide to work on that as opposed to what was agreed in the sprint planning session.
+The sprint planning works like this:
+- During the week of the sprint planning, teams (with the help of the product managers) define one or two big ambitious goals for this sprint that fit in with our roadmap.
```

Comment:
```suggestion
- During the week of the sprint planning, teams (with the help of the product managers) define one or two big ambitious goals for this sprint that fit in with [our roadmap](/handbook/strategy/roadmap).
```

Reply:
> If we don’t achieve the ambitious goal this sprint, can we have as a goal the next sprint a less ambitious “actually finish that goal” sprint? Won’t be burn out if we’re always aiming too high and never reaching the target? E.g. I find that I get motivated by goals with deadlines, and tend to push myself harder than it might make sense.

It's important to get this right, the way I see it theres a few different possible outcomes:
* **Nailed it:** We hit the goal spectacularly (high fives all round)
* **Scraped it:** We almost hit the goal - but we'll need to do a little bit more next sprint to tidy up (this likely falls into the 1/3rd below) - no need to burn ourselves out to Nail it if we think we're going to scrape it.
* **Failed it:** We were no-where near hitting the goal, but we learned some valuable lessons - we're going to go back to the drawing board - maybe the goal wasn't right or maybe there's a different way to approach it


>I think there should be some general rule regarding the capacity of the team that’s supposed to be dedicated towards the big 1-2 goals of the sprint, and what’s going to go for other work “any other work (smaller features, bug fixes, refactoring etc)“. Some general rule like “2/3 of the team’s capacity should go towards the big goal, 1/3 towards other work, split by people/days as needed” could be a good guide, which we can adjust according to experience.

2/3rds on big goal feels about right to me

Reply:
> **Scraped it:** We almost hit the goal - but we'll need to do a little bit more next sprint to tidy up (this likely falls into the 1/3rd below) - no need to burn ourselves out to Nail it if we think we're going to scrape it.

I'd strongly advise against putting this in the 1/3. Unplanned work is a real thing. The scenario where we're scrambling to finish what we did last sprint, haven't even started on the new sprint's big hairy audacious goals that we're never going to reach, and have a lot of inbound unplanned work, sounds like burnout, and sounds like the opposite of reducing WIP. We could compensate by having just one big goal the next sprint, or temporarily changing the allocation from "2/3 and 1/3" to "1/3 and 2/3" or 50%-50%.

Basically, just taking on a little bit less next sprint to finish the thing. Let's call it accordion-style planning.

Reply:
+1 to limiting work in progress and adjusting the allocation following a "scraped it"

Reply:
As a little bit of extra context, I've found that people often misunderstand what "done" means. Quoting from a (paid) newsletter I'm reading:

> Discuss and document what “definition of done” means. When is a feature considered complete? Is it when:
> 
> 1. The code compiles and the engineer did one test, and it worked? 
> 
> 2. After all edge cases are verified and have unit tests exercising them, the code passed code review, has been deployed to production, rolled out gradually, and there are no regressions noted, nor customer bug reports?
> 
> Many inexperienced engineers assume their work is done in the case of 1. However, at most places, the acceptance criteria is closer to 2 in complexity, and all teams have different additional quality expectations.

Let's be mindful of the  `diff` between cases 1 and 2, and let's not over-plan big goals when we're at the "it's almost done" stage. In my experience, there's a good chance you're 50% way done when you feel like "almost done", taking into account this `diff`.

Reply:
Thanks @mariusandra great point - I'll try and weave the outcome of this discussion into the doc tomorrow morning.

Reply:
Done - let me know if you have any further thoughts